CMakeLists.txt

```diff
@@ -195,6 +202,9 @@ option(
   OFF)
 option(WITH_EXAMPLES "Whether to build examples" ON)
 
+# This requires CURL, OFF by default.
+option(WITH_EXAMPLES_HTTP "Whether to build http client/server examples" OFF)
```

Comment:
Do you think it's better to set `WITH_EXAMPLES_HTTP` to `ON` when `WITH_HTTP_CLIENT_CURL` is `ON` or CURL is found?

Reply:
@owent 

Thanks for the comments.

The dependency is:

- WITH_EXAMPLES_HTTP
  - requires WITH_HTTP_CLIENT_CURL
    - requires CURL

so the configuration is evaluated from the top (what the user wants) to the bottom (what low level packages are needed to satisfy that).

What you suggest is to take the opposite route (bottom up), and to build some parts (WITH_EXAMPLES_HTTP) even when the user did not asked for it in the makefile, just because it is possible (CURL present)

The bottom up approach leads to unpredictable behavior:

- if another part uses CURL for something else, why would suddenly the http examples be built ?
- WITH_ZIPKIN will build http client/server examples as a side effect ?

Taking the bottom up approach, unpredictable items could be compiled, then packaged, and then installed, even when nobody asked for them.

I think the top down logic is more robust.

Reply:
Thanks for your explanation. In my understanding, if we set `WITH_EXAMPLES` to `ON` by default, I will espect all examples will be built.
That's fine by me to exclude http examples by default, I just thought it may lead to some misunderstanding.